packages/fx-core/src/plugins/resource/bot/v2/common.ts

```diff
@@ -78,3 +78,14 @@ export function resolveServiceType(ctx: Context): ServiceType {
     ] as string) ?? HostType.AppService;
   return getServiceType(rawHostType);
 }
+
+export function resolveBotCapabilities(inputs: Inputs): BotCapability[] {
+  const botScenarios = inputs?.[AzureSolutionQuestionNames.Scenarios];
+  if (Array.isArray(botScenarios)) {
+    return botScenarios
+      .map((scenario) => QuestionBotScenarioToBotCapability.get(scenario))
+      .filter((item): item is BotCapability => item !== undefined);
```

Comment:
The code is hard to understand. How about changing it to 
```
return botScenarios
      .filter(it => QuestionBotScenarioToBotCapability.has(it))
      .map((scenario) => QuestionBotScenarioToBotCapability.get(scenario))
```

Reply:
Even if filter it first, the return type of get method is still `BotCapability|undifined` and still need a second filter. But indeed it;s hard to understand. I will change it to get()! here.